core/txn/pool/gatherer.go

```diff
@@ -41,18 +93,28 @@ type item struct {
 	ch  chan []txn.Transaction
 }
 
+// SimpleGatherer is a gatherer of transactions that will use filters to drop
+// invalid transactions. It limits the size for each identity, *as long as* a
+// filter is set, otherwise it can grow indefinitely.
+//
+// - implements pool.Gatherer
 type simpleGatherer struct {
 	sync.Mutex
-	queue   []item
-	set     map[Key]txn.Transaction
-	history map[Key]struct{}
+
+	limit      int
+	queue      []item
+	validators []Filter
+
+	// A string key is generated for each unique identity, which will have its
+	// own list of transactions.
+	txs map[string]transactions
```

Comment:
For extra clarity I would explain why we need to store the Txs by identities, instead of having a simple slice or bag of Txs. This is something I'm wondering now